### PR TITLE
Staged replacement of the use of evaluate* and get* with evaluate() and value() [3/N]

### PR DESCRIPTION
**Pull request by protonu**
Replaces uses of evaluate(Int, Bool, Double) with evaluate which returns a PolymorphicValue.

This also replaces the uses of get(Int, Bool, Double) with value().

This PR is the third of a number of them to fix the issue. In the final PR, the older get* and evaluate* methods will be removed.

Fixes https://github.com/NVIDIA/Fuser/issues/643

### CODE REVIEW

**Comment by protonu** [comment]
!build

**Comment by protonu** [comment]
!build

File: csrc/device_lower/validation.cpp [hunk 1]
```diff
@@ -1192,7 +1191,8 @@ void validateAndConvertIterDomainGrouping(Fusion* fusion) {
 
       // Extent must be static
       NVF_CHECK(
-          id->extent()->getInt().has_value(),
+          (id->extent()->value().hasValue() &&
+           id->extent()->value().is<int64_t>()),
```
Comment (zasdfgbnm) [inline csrc/device_lower/validation.cpp:1195]:
```suggestion
          id->extent()->value().is<int64_t>(),
```

File: csrc/ops/arith.cpp [hunk 2]
```diff
@@ -1493,8 +1495,8 @@ TensorView* expand(TensorView* inp, const std::vector<Val*>& expanded_sizes) {
       // Input id is non-expand and its extent is concrete. Nothing
       // to expand, but the input and expanded sizes should match if
       // the expanded size is also concrete.
-      auto inp_id_size_int = inp_id->extent()->evaluateInt();
-      if (expanded_size_int.has_value()) {
+      auto inp_id_size_int = inp_id->extent()->evaluate();
+      if (expanded_size_int.hasValue() && expanded_size_int.is<int64_t>()) {
```
Comment (zasdfgbnm) [inline csrc/ops/arith.cpp:1499]:
```suggestion
      if (expanded_size_int.is<int64_t>()) {
```

File: csrc/device_lower/analysis/predicate_elimination.cpp [hunk 3]
```diff
@@ -153,16 +153,15 @@ class PredicateAnalyzer : public OptOutDispatch {
   // axis. Otherwise, we can't skip predication as it might cause
   // out-bound accesses with the producer tensor
   void handle(Split* split) override {
-    auto factor = split->factor()->getInt();
-    if (!factor.has_value()) {
+    auto factor = split->factor()->value();
+    if (!factor.hasValue() || !factor.is<int64_t>()) {
```
Comment (zasdfgbnm) [inline csrc/device_lower/analysis/predicate_elimination.cpp:157]:
```suggestion
    if (!factor.is<int64_t>()) {
```

**Comment by protonu** [comment]
!build

File: tools/cuda-keyring_1.0-1_all.deb
Comment (zasdfgbnm) [inline tools/cuda-keyring_1.0-1_all.deb:1]: Please remove before merge

File: tools/cuda-keyring_1.0-1_all.deb.1
Comment (zasdfgbnm) [inline tools/cuda-keyring_1.0-1_all.deb.1:1]: Please remove before merge

**Review by zasdfgbnm — APPROVED** [review]

File: tools/cuda-keyring_1.0-1_all.deb
Comment (protonu) [reply tools/cuda-keyring_1.0-1_all.deb:1]: Will do!

File: tools/cuda-keyring_1.0-1_all.deb.1
Comment (protonu) [reply tools/cuda-keyring_1.0-1_all.deb.1:1]: Same here!!

**Comment by protonu** [comment]
!build

**Comment by protonu** [comment]
!build

**Comment by protonu** [comment]
!build

**Comment by xwang233** [comment]
@zasdfgbnm do you think the cpp build failures are relevant?

**Comment by zasdfgbnm** [comment]
> @zasdfgbnm do you think the cpp build failures are relevant?

This doesn't look related, but why it is only failing on this PR?

**Comment by zasdfgbnm** [comment]
!build

**Comment by protonu** [comment]
!build

**Comment by protonu** [comment]
!build

**Comment by protonu** [comment]
!build

**Pull request merged**